Java/syntax_test_java.java

```diff
@@ -311,4 +311,30 @@ void test2() {
 //      ^ storage.modifier
 //            ^ support.class
     }
+
+    @Partial @Mock(type=Grafo.class) DataLoader inline;
+//  ^^^^^^^^                          meta.annotation
+//           ^^^^^^^^^^^^^^^^^^^^^^^ meta.annotation
+//                                   ^ support.class
+
+    @Override public int inline() {
+//  ^^^^^^^^^ meta.annotation
+//  ^ punctuation.definition.annotation
+//   ^^^^^^^^ variable.annotation
+//            ^ meta.method
+    }
+
+    void annotatedArgs(@NonNull final String p1,
+//                     ^^^^^^^^ meta.annotation
+//                              ^ storage.modifier - meta.annotation
+        @Named(value = "") List<T> p2, @NonNull final String p3) {}
+//      ^^^^^^^^^^^^^^^^^^ meta.annotation
+//                         ^ support.class
+//                             ^ meta.generic punctuation.definition.generic.begin
+//                                 ^ variable.parameter
+//                                   ^ punctuation.separator
+//                                     ^^^^^^^^ meta.annotation
+//                                              ^ storage.modifier - meta.annotation
+//                                                    ^ support.class
```

Comment:
I also tested this manually, but I don't think it'll benefit to add it to the test suite, since the above test case is complicated enough as is:

```java
        @Named(foo = "", bar = BAZ) List<T> p2, @NonNull final String p3) {}
```

although not included here, I also manually verified that the punctuation in the method args list is detected properly even when there are commas in the annotation's parameter list.